nixos/modules/services/misc/gollum.nix

```diff
@@ -0,0 +1,96 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.gollum;
+in
+
+{
+  options.services.gollum = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Enable the Gollum service.";
+    };
+
+    package = mkOption {
+      default = pkgs.gollum;
+      defaultText = "pkgs.gollum";
+      description = "Gollum package used for the service";
+    };
+
+    address = mkOption {
+      type = types.str;
+      default = "0.0.0.0";
+      description = "IP address on which the web server will listen.";
+    };
+
+    port = mkOption {
+      type = types.int;
+      default = 4567;
+      description = "Port on which the web server will run.";
+    };
+
+    extraConfig = mkOption {
+      type = types.str;
+      default = "";
+      description = "Content of the configuration file";
+    };
+
+    branch = mkOption {
+      type = types.str;
+      default = "master";
+      example = "develop";
+      description = "Git branch to serve";
+    };
+
+    stateDir = mkOption {
+      type = types.path;
+      default = "/var/lib/gollum";
+      description = "Specifies the path of the repository directory. If it does not exist, Gollum will create it on startup.";
+    };
+
+  };
+
+  config = mkIf cfg.enable {
+
+    users.extraUsers.gollum = {
+      group = config.users.users.gollum.name;
+      description = "Gollum user";
+      createHome = false;
+      uid = config.ids.uids.gollum;
+    };
+
+    users.extraGroups.gollum.gid = config.ids.gids.gollum;
+
+    systemd.services.gollum = {
+      description = "Gollum wiki";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      path = [ pkgs.git ];
+
+      preStart = ''
+        # All of this is safe to be run on an existing repo
+        mkdir -p ${cfg.stateDir}
+        git init ${cfg.stateDir}
+        chmod 755 ${cfg.stateDir}
+        chown -R gollum:gollum ${cfg.stateDir}
```

Comment:
Fixing ownership at startup seems to make allocating a static uid redundant. If a static uid must be used, this is at least a waste of time.

Reply:
Thanks! This part was meant as a temporarily solution I forgot to fix. I'd like to let users the choice of renaming the `gollum` user. This means, that line 78 is supposed to look like this:

```
chown -R ${cfg.user}:${cfg.group} ${cfg.stateDir}
```

If you think, that with this solution I would'nt need a static UID/GID, I will remove them.

Reply:
With fixing the ownership, I expect a static uid is unnecessary. If updating the ownership is very expensive, however, perhaps stick with only static uid. Then again, I don't really know the software so I'll leave that to your judgement.

Reply:
Now I think I got what you said. I removed the static UID/GID. That generic userName/groupName feature i mentioned will be added later, as I don't see immediate need for it.